compiler/rustc_passes/src/liveness.rs

```diff
@@ -1199,7 +1199,7 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
                             }
                         }
                         hir::InlineAsmOperand::InOut { expr, .. } => {
-                            succ = self.propagate_through_place_components(expr, succ);
+                            succ = self.propagate_through_expr(expr, succ);
```

Comment:
I didn't dig too deep into this, but taking `Assign` code as an example or `SplitInOut`, I think this may need both `propagate_through_place_components` and `propagate_through_expr`.

Reply:
It looks like it was determined that `propagate_through_place_components` would be the necessary call here in order to avoid improperly setting the inout expr as its own successor/predecessory.